From 42bb4d896d3441b16604823cb57c00bdd37ca593 Mon Sep 17 00:00:00 2001
From: OpenEmbedded <oe.patch@oe>
Date: Thu, 28 Nov 2024 12:39:37 +0100
Subject: [PATCH] Add sy20257 regulator

---
 drivers/regulator/sy8824x.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/regulator/sy8824x.c b/drivers/regulator/sy8824x.c
index 5e915cf307b3..f73f57f87727 100644
--- a/drivers/regulator/sy8824x.c
+++ b/drivers/regulator/sy8824x.c
@@ -183,6 +183,16 @@ static const struct sy8824_config sy8824e_cfg = {
 	.config = &sy8824_regmap_config,
 };
 
+static const struct sy8824_config sy20257n_cfg = {
+	.vol_reg = 0x00,
+	.mode_reg = 0x00,
+	.enable_reg = 0x00,
+	.vsel_min = 600000,
+	.vsel_step = 12500,
+	.vsel_count = 64,
+	.config = &sy8824_regmap_config,
+};
+
 static const struct sy8824_config sy20276_cfg = {
 	.vol_reg = 0x00,
 	.mode_reg = 0x01,
@@ -212,6 +222,10 @@ static const struct of_device_id sy8824_dt_ids[] = {
 		.compatible = "silergy,sy8824e",
 		.data = &sy8824e_cfg
 	},
+	{
+		.compatible = "silergy,sy20257n",
+		.data = &sy20257n_cfg
+	},
 	{
 		.compatible = "silergy,sy20276",
 		.data = &sy20276_cfg
